transport_test.go

```diff
@@ -198,32 +198,6 @@ func TestGetRequestFromEvent(t *testing.T) {
 	}
 }
 
-func TestRetryAfterNoHeader(t *testing.T) {
-	assertEqual(t, retryAfter(time.Now(), nil), time.Second*60)
```

Comment:
👍 

Have to add it to the transport implementations, we didn't have any tests in that direction I think.
I've sanity checked manually with a small modification to `examples/basic/main.go` and a reduced quota for errors in a test project.

Reply:
Done.

These tests that were apparently deleted were actually moved to `retry_after_test.go` with a few additions (and that was in the first iteration of the PR).

(new) Added `TestRateLimiting` below to exercise both transports shipped with the SDK against a test server that simulates rate limits.

The new tests helped uncover at least two bugs: 1) I forgot to merge instead of replace rate limits in `HTTPTransport`; 2) Data race in `HTTPSyncTransport`.

Thank you, it was worth the addition!